mlir/include/mlir/Dialect/XeGPU/IR/XeGPUOps.td

```diff
@@ -163,9 +176,19 @@ def XeGPU_CreateNdDescOp: XeGPU_Op<"create_nd_tdesc", [Pure, ViewLikeOpInterface
     }
 
     ArrayRef<int64_t> getStaticOffsets(){
-      return getConstOffsets();
+      auto attr = getConstOffsetsAttr();
+      if (llvm::isa<IntegerType>(getSourceType()) || attr)
+        return attr;
+
+      // The offsets are allowed to be empty. The Traits verification of OffsetSizeAndStrideOpInterface interface assumes offsets being present. So it is set to be MAX to indicate user not passed any value (kDynamic means offsets passed as variable).
+      setConstOffsets(llvm::SmallVector<int64_t, 4>(getTensorDescShape().size(), std::numeric_limits<int64_t>::max()));
```

Comment:
Here using TensorDescShape as anchor is not appropriate, since TensorDesc could have smaller rank than MemRef. Use Shape for MemRef is better.